core/Cargo.toml

```diff
@@ -13,9 +13,15 @@ edition = "2018"
 cfg-if = "1.0.0"
 smallvec = "1.6.1"
 petgraph = { version = "0.6.0", optional = true }
-thread-id = { version = "4.0.0", optional = true }
 backtrace = { version = "0.3.60", optional = true }
 
+[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
+thread-id = { version = "4.0.0", optional = true }
+
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+js-sys = { version = "0.3.50", optional = true }
+web-sys = { version = "0.3.50", optional = true, features = ["console"] }
```

Comment:
The js-sys and web-sys crates are only for usage on the web when using wasm-bindgen. They don't work with wasm32-wasi, nor with wasm32-unknown-emscripten.

Reply:
Do you know what we can use for wasi & emecripten for getting a thread id / worker name?

Reply:
wasm32-wasi doesn't have threads. wasm32-wasi-threads will have them once added, but doesn't have thread names. For emscripten there is a way to call arbitrary javascript I believe, but I don't know how to do that from rust.